src/components/App.js

```diff
@@ -1,14 +1,45 @@
 import React from 'react';
 import Display from './Display';
 import ButtonPanel from './ButtonPanel';
+import calculate from '../logic/calculate';
 
-function App() {
-  return (
-    <div className="calculator">
-      <Display />
-      <ButtonPanel />
-    </div>
-  );
+class App extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      total: null,
+      next: null,
+      operation: null,
+    };
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick(buttonName) {
+    const { total, next, operation } = this.state;
+    this.setState(calculate({ total, next, operation }, buttonName));
+  }
+
+  render() {
+    const { total, next, operation } = this.state;
+    let result;
+
+    if (total) {
+      if (next) {
+        result = `${total}${operation}${next}`;
+      } else {
+        result = `${total}${operation}`;
+      }
+    } else {
+      result = operation ? `${next}${operation}` : next;
+    }
+
+    return (
+      <div className="calculator">
+        <Display result={result} />
```

Comment:
- Pass `total` or `next` value. You may do something like:
```javascript
<Display value={this.state.next || this.state.total || '0'} />
```

Reply:
Thanks for the suggestion. Done :heavy_check_mark: